### PR TITLE
Added tests for broadcast filter using regular expr

### LINKED ISSUE #34: Compatibility states python 3.5 but should fail
In `BroadcastFilter._ensure_filter` we find the following line:
```
if isinstance(filter_value, typing.re.Pattern):
```
which in python 3.5.2 will raise
```
TypeError: Type aliases cannot be used with isinstance().
```
Apparently this also goes for all older python3 version: https://github.com/pandas-dev/pandas/issues/21078
We should either replace it with a different check, or remove python 3.4 and 3.5 from compatibility list.

### PR DESCRIPTION
**Pull request by muhrin**
Fixes #34 

and direct match.  This makes sure all three paths through
_ensure_filter are hit.

Added py3.5 compatible regular expression type match

Thought I'd take this opportunity to fix up this issue.

### CODE REVIEW

File: test/utils.py [hunk 1]
```diff
@@ -347,6 +348,58 @@ def on_broadcast_1(_communicator, _body, sender=None, subject=None, _correlation
         self.assertEqual(4, len(senders_and_subects))
         self.assertSetEqual(EXPECTED, senders_and_subects)
 
+    def test_broadcast_filter_sender_and_subject_regex(self):
+        """As the standard broadcast test but using regular expressions instead of wildcards"""
+        # pylint: disable=invalid-name
+        senders_and_subects = set()
```
Comment (sphuber) [inline test/utils.py:354]:
```suggestion
        senders_and_subjects = set()
```

File: test/utils.py [hunk 2]
```diff
@@ -347,6 +348,58 @@ def on_broadcast_1(_communicator, _body, sender=None, subject=None, _correlation
...
+            ('bob.jones', 'purchase.car'),
+            ('bob.jones', 'purchase.piano'),
+            ('alice.jones', 'purchase.car'),
+            ('alice.jones', 'purchase.piano'),
+        }
+
+        done = kiwipy.Future()
+
+        def on_broadcast_1(_communicator, _body, sender=None, subject=None, _correlation_id=None):
+            senders_and_subects.add((sender, subject))
```
Comment (sphuber) [inline test/utils.py:365]:
```suggestion
            senders_and_subjects.add((sender, subject))
```

File: test/utils.py [hunk 3]
```diff
@@ -347,6 +348,58 @@ def on_broadcast_1(_communicator, _body, sender=None, subject=None, _correlation
...
+            if len(senders_and_subects) == len(EXPECTED):
+                done.set_result(True)
+
+        filtered = kiwipy.BroadcastFilter(on_broadcast_1)
+        filtered.add_sender_filter(re.compile(".*.jones"))
+        filtered.add_subject_filter(re.compile("purchase.*"))
+        self.communicator.add_broadcast_subscriber(filtered)
+
+        for sendr in ['bob.jones', 'bob.smith', 'martin.uhrin', 'alice.jones']:
```
Comment (sphuber) [inline test/utils.py:374]:
```suggestion
        for sender in ['bob.jones', 'bob.smith', 'martin.uhrin', 'alice.jones']:
```

File: test/utils.py [hunk 4]
```diff
@@ -347,6 +348,58 @@ def on_broadcast_1(_communicator, _body, sender=None, subject=None, _correlation
...
+        done = kiwipy.Future()
+
+        def on_broadcast_1(_comm, _body, sender=None, _subject=None, _correlation_id=None):
+            senders.append(sender)
+            if len(senders) == len(EXPECTED_SENDERS):
+                done.set_result(True)
+
+        self.communicator.add_broadcast_subscriber(kiwipy.BroadcastFilter(on_broadcast_1, sender="alice.jones"))
+
+        for sendr in ['bob.jones', 'bob.smith', 'martin.uhrin', 'alice.jones']:
```
Comment (sphuber) [inline test/utils.py:396]:
```suggestion
        for sender in ['bob.jones', 'bob.smith', 'martin.uhrin', 'alice.jones']:
```

**Comment by muhrin** [comment]
Turns out me no writie so good

**Pull request merged**